### PR TITLE
Bug24104 029

### PR DESCRIPTION
**Pull request by juga0**
As #131 but based on 0.2.9

### CODE REVIEW

**Comment by juga0** [comment]
Is it needed a changes file?
There is no test for these changes yet, should them be part of the same ticket?

File: changes/bug24104 [hunk 1]
```diff
@@ -0,0 +1,4 @@
+  o Minor bugfix (relay statistics):
+    - Update relay descriptor on bandwidth changes only when the uptime is
+      bigger than 24h in order to reduce the efficiency of guard discovery
+      attacks. Fixes bug 24104; bugfix on 0.1.1.6-alpha.
```
Comment (ahf) [inline changes/bug24104]: Unsure if our script can handle a lack of trailing newline here?

File: src/or/router.c [hunk 2]
```diff
@@ -2425,21 +2425,35 @@ mark_my_descriptor_dirty(const char *reason)
...
   if (!router_get_my_routerinfo())
     return;
 
   prev = router_get_my_routerinfo()->bandwidthcapacity;
   cur = we_are_hibernating() ? 0 : rep_hist_bandwidth_assess();
   if ((prev != cur && (!prev || !cur)) ||
-      cur > prev*2 ||
-      cur < prev/2) {
+      cur > prev * BANDWIDTH_CHANGE_FACTOR ||
+      cur < prev / BANDWIDTH_CHANGE_FACTOR ) {
```
Comment (ahf) [inline src/or/router.c]: Very minor style issue: put () around the expression on the right-hand side of the < and >

File: src/test/log_test_helpers.c [hunk 3]
```diff
@@ -158,6 +158,26 @@ mock_saved_log_has_message_containing(const char *msg)
...
+/**
+ * Return true iff there is not a message recorded by log capture
+ * that contains <b>msg</b> as a substring.
+ */
+int
+mock_saved_log_has_message_not_containing(const char *msg)
+{
+  if (saved_logs) {
+    SMARTLIST_FOREACH(saved_logs, mock_saved_log_entry_t *, m,
+                      {
```
Comment (ahf) [inline src/test/log_test_helpers.c]: Minor style issue: the indentation looks a bit off here?

File: src/test/test_router.c [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+static int
+mock_we_are_hibernating(void)
+{
+  return 0;
+}
+
+static void
+test_router_check_descriptor_bandwidth_changed(void *arg)
+{
+
```
Comment (ahf) [inline src/test/test_router.c]: Remove empty line

File: src/test/log_test_helpers.c (hunk 3 above)
Comment (juga0) [reply src/test/log_test_helpers.c]: indeed is odd, but the code around uses the same identation

**Comment by juga0** [comment]
Closing this and continuing in #241, cause compared with master by mistake

**Comment by juga0** [comment]
s/#241/#189

**Pull request closed without merging**